scaleway/helpers_secret.go

```diff
@@ -35,3 +36,27 @@ func secretAPIWithRegionAndID(m interface{}, id string) (*secret.API, scw.Region
 	}
 	return api, region, id, nil
 }
+
+// secretVersionAPIWithRegionAndID returns a Secret API with locality and Nested ID extracted from the state
+func secretVersionAPIWithRegionAndID(m interface{}, id string) (*secret.API, scw.Region, string, string, error) {
+	meta := m.(*Meta)
+	api := secret.NewAPI(meta.scwClient)
+
+	region, id, revision, err := parseLocalizedNestedID(id)
+	if err != nil {
+		return nil, "", "", "", err
+	}
+	return api, scw.Region(region), id, revision, nil
```

Comment:
```suggestion

	region, id, revision, err := parseLocalizedNestedID(id)
	if err != nil {
		return nil, "", "", "", err
	}

	api := secret.NewAPI(meta.scwClient)
	return api, scw.Region(region), id, revision, nil
```

Avoid allocating memory if it may be not used